src/EventuallyQueue.js

```diff
@@ -276,7 +279,7 @@ const EventuallyQueue = {
         await object.save(queueObject.object, queueObject.serverOptions);
         await this.remove(queueObject.queueId);
       } catch (e) {
-        if (e.code !== ParseError.CONNECTION_FAILED) {
+        if (![ParseError.CONNECTION_FAILED, ParseError.INVALID_SESSION_TOKEN].includes(e.code)) {
```

Comment:
If I read this correctly, pre-PR, an operation in the queue will be discarded if the session token is invalid; post-PR, the operation stays in the queue.
- What type of queue is this? Will a such a pending operation block the queue for others to be executed?
- When will these failed operations due to invalid session token be discarded? After n failed retries?
- How is the queue managed when the current user logs out - will the queue emptied?
- How is the queue managed when another user logs in - will the queue be emptied?

Reply:
> What type of queue is this? Will a such a pending operation block the queue for others to be executed?
A standalone non-blocking queue

> When will these failed operations due to invalid session token be discarded? After n failed retries?
If a valid sessionToken exists and the save fails for another reason, then it will be discarded

> How is the queue managed when the current user logs out - will the queue emptied?
Currently the queue isn't cleared on logout

> How is the queue managed when another user logs in - will the queue be emptied?
The queue runs when a user logs in

Reply:
Thanks for the insight, I've invites you to a separate thread to discussion.